src/lib/general.c

```diff
@@ -59,7 +59,7 @@ static void parse_lib_version(CK_BYTE *major, CK_BYTE *minor) {
 
     endptr = NULL;
     val = strtoul(minor_str, &endptr, 10);
-    if (errno != 0 || endptr[0] || val > UINT8_MAX) {
```

Comment:
I think the logic needs to be fixed around line 38, null terminate the string on the next `.` or `-` character if found.

Reply:
Not tested at all, but offhand I am thinking something like this:
```diff
diff --git a/src/lib/general.c b/src/lib/general.c
index eaddaf82a97a..91c3d841a21d 100644
--- a/src/lib/general.c
+++ b/src/lib/general.c
@@ -38,6 +38,13 @@ static void parse_lib_version(CK_BYTE *major, CK_BYTE *minor) {
     if (split) {
         split[0] = '\0';
         minor_str = &split[1];
+        split = strchr(minor_str, '.');
+        if (!split) {
+            split = strchr(minor_str, '-');
+        }
+        if (split) {
+            split[0]='\0';
+        }
     }
 
     if (!major_str[0] || !minor_str[0]) {
```